### PR TITLE
Adjusted test cases to verify all functionality in hatrac.js

### PR DESCRIPTION
**Pull request by jrchudy**
This PR addresses issue #424.

 - The changes to `hatrac.js` were made to collect all private functions in one place and to also redefine those private functions with a leading `_`. 
 - The changes to `01.checksum.js` were made to verify the values of the checksum object.
 - Rewrote `02.*.js` to test the functionality of `hatrac.js` completely rather than bits and pieces of the functionality.

I made sure my changes did not break any chaise e2e tests.

### CODE REVIEW

File: test/specs/upload/tests/02.upload_obj.js [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+                }
+
+                done();
+            }, function (err) {
+                console.dir(err);
+                done.fail();
+            });
+        });
+
+        it("should have properties set appropriately by it's constructor.", function () {
```
Comment (RFSH) [inline test/specs/upload/tests/02.upload_obj.js]: Typo "it's"

File: test/specs/upload/tests/02.upload_obj.js [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+                expect(uploadObj.jobDone).toBeTruthy("Upload job is not complete");
+
+                done();
+            }).catch(function(err) {
+                console.dir(err);
+                done.fail();
+            });
+        });
+
+        // TODO: Upload.pause(), Upload.resume(), and Upload.cancel() are not tested.
```
Comment (RFSH) [inline test/specs/upload/tests/02.upload_obj.js]: Can you add some comments that why we are not testing these function now, and how we might be able to test them later?

**Comment by jrchudy** [comment]
@RFSH I updated `hatrac.js` with a function to delete files from hatrac. I couldn't define it as a private method because then the tests couldn't use it..

This means we should be able to call this function from chaise to delete files created by chaise tests.

**Comment by RFSH** [comment]
Thanks. I will use this function in chaise test cases to remove the files. You can go ahead and merge this PR.

**Pull request merged**